spark/src/main/java/org/opensearch/sql/spark/cluster/FlintStreamingJobCleanerTask.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.cluster;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.sql.datasource.DataSourceService;
+import org.opensearch.sql.datasource.model.DataSourceMetadata;
+import org.opensearch.sql.datasource.model.DataSourceStatus;
+import org.opensearch.sql.datasource.model.DataSourceType;
+import org.opensearch.sql.legacy.metrics.MetricName;
+import org.opensearch.sql.legacy.metrics.Metrics;
+import org.opensearch.sql.spark.client.EMRServerlessClientFactory;
+import org.opensearch.sql.spark.dispatcher.model.FlintIndexOptions;
+import org.opensearch.sql.spark.execution.statestore.StateStore;
+import org.opensearch.sql.spark.flint.FlintIndexMetadata;
+import org.opensearch.sql.spark.flint.FlintIndexMetadataService;
+import org.opensearch.sql.spark.flint.operation.FlintIndexOpAlter;
+import org.opensearch.sql.spark.flint.operation.FlintIndexOpCancel;
+
+/** Cleaner task which alters the active streaming jobs of a disabled datasource. */
+@RequiredArgsConstructor
+public class FlintStreamingJobCleanerTask implements Runnable {
+
+  private final DataSourceService dataSourceService;
+  private final FlintIndexMetadataService flintIndexMetadataService;
+  private final StateStore stateStore;
+  private final EMRServerlessClientFactory emrServerlessClientFactory;
+
+  private static final Logger LOGGER = LogManager.getLogger(FlintStreamingJobCleanerTask.class);
+  protected static final AtomicBoolean isRunning = new AtomicBoolean(false);
+
+  @Override
+  public void run() {
+    if (!isRunning.compareAndSet(false, true)) {
+      LOGGER.info("Previous task is still running. Skipping this execution.");
+      return;
+    }
+    try {
+      LOGGER.info("Starting the cleaner task for disabled and deleted data sources.");
+      List<DataSourceMetadata> s3GlueDisabledDataSources = getS3GlueDataSources();
+      Set<String> disabledS3DataSources =
+          s3GlueDisabledDataSources.stream()
+              .filter(
+                  dataSourceMetadata -> dataSourceMetadata.getStatus() == DataSourceStatus.DISABLED)
```

Comment:
datasource could in enable state after search operation. could we check datasource enabled in spark job health check logic?

Reply:
If we could do that, we don't need this cron at all. We can add only datasource enabled check in streaming job itself.

Reply:
If we stick with CRON, since we run the step again after 15min. The datasource will be enabled.

Reply:
> If we stick with CRON, since we run the step again after 15min. The datasource search would give disable status.

my original concern is
T1, user disable datasource.
T2, cron job scan disabled datasource. 
T3, user enable datasource.
T4, cron cancel all the job related to datasource.
it seems works fine, because user disable datasource T1.

my only concern of validation in Spark side is dependency on datasource state.

Reply:
One solution is I could query datasource details for every index individually right before altering 
rather than querying  all the datasource metadata upfront.

Do you see any solutions for this?